server/src/main/java/org/elasticsearch/action/admin/indices/flush/SyncedFlushResponse.java

```diff
@@ -124,6 +156,160 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
+                                            ensureExpectedToken(Token.FIELD_NAME, curToken, parser::getTokenLocation);
+                                            String level3FieldName = parser.currentName();
+                                            curToken = parser.nextToken();
+                                            switch (level3FieldName) {
+                                                case Fields.SHARD:
+                                                    ensureExpectedToken(Token.VALUE_NUMBER, curToken, parser::getTokenLocation);
+                                                    shardId = new ShardId(
+                                                        fieldName,
+                                                        IndexMetaData.INDEX_UUID_NA_VALUE,
+                                                        parser.intValue()
+                                                    );
+                                                    break;
+                                                case Fields.REASON:
+                                                    ensureExpectedToken(Token.VALUE_STRING, curToken, parser::getTokenLocation);
+                                                    failureReason = parser.text();
+                                                    break;
+                                                case Fields.TOTAL_COPIES:
+                                                    ensureExpectedToken(Token.VALUE_NUMBER, curToken, parser::getTokenLocation);
+                                                    totalCopies = parser.intValue();
+                                                    break;
+                                                case Fields.SUCCESSFUL_COPIES:
+                                                    ensureExpectedToken(Token.VALUE_NUMBER, curToken, parser::getTokenLocation);
+                                                    successfulCopies = parser.intValue();
+                                                    break;
+                                                case Fields.ROUTING:
+                                                    routing = ShardRouting.fromXContent(parser);
+                                                    break;
+                                                default:
+                                                    // If something else skip it
+                                                    parser.skipChildren();
+                                                    break;
+                                            }
+                                        }
+                                        if (failureReason != null &&
+                                            shardId != null &&
+                                            totalCopies != null &&
+                                            successfulCopies != null) {
```

Comment:
Yes. Same reason as above. I have a general tendency of always expecting malformed JSONs I think. This would mean that if someone changed the `toXContent` method tests would start to fail. I will remove it if you think it is not required.